src/lib/core/datetime/calendar-locale.spec.ts

```diff
@@ -0,0 +1,90 @@
+import {inject, TestBed, async} from '@angular/core/testing';
+import {CalendarLocale} from './calendar-locale';
+import {DatetimeModule} from './index';
+import {SimpleDate} from './simple-date';
+
+
+describe('DefaultCalendarLocale', () => {
+  let calendarLocale: CalendarLocale;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [DatetimeModule],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(inject([CalendarLocale], (cl: CalendarLocale) => {
+    calendarLocale = cl;
+  }));
+
+  it('lists months', () => {
+    expect(calendarLocale.months).toEqual([
+      'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September',
+      'October', 'November', 'December'
+    ]);
+  });
+
+  it('lists short months', () => {
+    expect(calendarLocale.shortMonths).toEqual([
+      'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'
+    ]);
+  });
+
+  it('lists narrow months', () => {
+    expect(calendarLocale.narrowMonths).toEqual([
+      'J', 'F', 'M', 'A', 'M', 'J', 'J', 'A', 'S', 'O', 'N', 'D'
+    ]);
+  });
+
+  it('lists days', () => {
+    expect(calendarLocale.days).toEqual([
+      'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'
+    ]);
+  });
+
+  it('lists short days', () => {
+    expect(calendarLocale.shortDays).toEqual(['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']);
+  });
+
+  it('lists narrow days', () => {
+    expect(calendarLocale.narrowDays).toEqual(['S', 'M', 'T', 'W', 'T', 'F', 'S']);
+  });
+
+  it('lists dates', () => {
+    expect(calendarLocale.dates).toEqual([
+      null, '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16',
```

Comment:
There's a comment in the base class, basically I thought it would be nice to not have to do `calendarLocale.dates[date + 1]` all the time (especially since months and days are 0-indexed), so I added `null` as the representation for `dates[0]`